src/app/core/services/machine-deployment.ts

```diff
@@ -82,4 +91,46 @@ export class MachineDeploymentService {
     const url = `${this._restRoot}/projects/${projectID}/clusters/${cluster}/machinedeployments/${mdId}/nodes/events`;
     return this._httpClient.get<Event[]>(url);
   }
+
+  showExternalMachineDeploymentDeleteDialog(
```

Comment:
Should we create a separate service for external clusters, similar to how we have `NodeService`? And then move this method (and upcoming edit machine deployment method) there? All other methods are API related in this service so this becomes an odd one out.

Reply:
I actually have a ticket  https://github.com/kubermatic/dashboard/issues/4655 and even placing method here or in ClusterServices where other methods related to external clusters are placed.
 
 Actually there should be 
 - MachineDeploymentService ( Already exists -> and currently, handling both KKP managed clusters + External cluster methods too)
- External MachineDeploymentService (Which should handles all external cloud provider supported e.g eks, aks, gke, kubeone(future)


I don't want to have big changeset but 1 thing we can do is move this method from `src/app/core/services/machine-deployment.ts` to NEW service call src/app/core/services/external-machine-deployment.ts (Service) and later on we can refactor other things

thoughts?

Reply:
Sounds good